packages/esm-patient-orders-app/src/results/result-form-field.component.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import styles from './result-form.scss';
+import { TextInput, Select, SelectItem } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { type ConceptReference } from './result-form.resource';
+import { Controller } from 'react-hook-form';
+
+interface ResultFormFieldProps {
+  defaultValue: any;
+  concept: ConceptReference;
+  control: any;
+  register: any;
+  errors?: any;
+}
+const ResultFormField: React.FC<ResultFormFieldProps> = ({ defaultValue, register, concept, control, errors }) => {
+  const { t } = useTranslation();
+  const isTextOrNumeric = (concept) => concept.datatype?.display === 'Text' || concept.datatype?.display === 'Numeric';
+  const isCoded = (concept) => concept.datatype?.display === 'Coded';
+  const isPanel = (concept) => concept.setMembers?.length > 0;
+
+  const printValueRange = (concept: ConceptReference) => {
+    if (concept?.datatype?.display === 'Numeric') {
+      const maxVal = Math.max(concept?.hiAbsolute, concept?.hiCritical, concept?.hiNormal);
+      const minVal = Math.min(concept?.lowAbsolute, concept?.lowCritical, concept?.lowNormal);
+      return ` (${minVal ?? 0} - ${maxVal > 0 ? maxVal : '--'} ${concept?.units ?? ''})`;
```

Comment:
We'll need a good way to show the ranges. For some tests, not all the ranges are defined, and getting the max of all the highs and min of the lows sometimes doesn't yield the required output. We may also need to evaluate whether the ranges should be displayed at this point.

Reply:
Thanks @ojwanganto for the feedback. Did u have something in mind?